src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,177 @@
+package duke.parser;
+
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.tasklist.TaskList;
+import duke.exception.DukeException;
+import duke.ui.Ui;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Parser class is responsible for interpreting user input and converting it into
+ * commands that can be executed by the Duke application. It also handles parsing
+ * of task data from and to the storage format.
+ */
+public class Parser {
+    
+    /**
+     * Parses the input command to identify the task to be marked as done.
+     *
+     * @param input The user input string.
+     * @param list The TaskList containing all tasks.
+     * @return The Task to be marked as done.
+     */
+    public Task getTaskTobeMarked(String input, TaskList list) {
+        int index = Integer.parseInt(input.substring(5)) - 1;
+        return list.getTask(index);
+    }
+    
+    /**
+     * Parses the input command to identify the task to be unmarked (marked as not done).
+     *
+     * @param input The user input string.
+     * @param list The TaskList containing all tasks.
+     * @return The Task to be unmarked.
+     */
+    public Task getTaskToBeUnmarked(String input, TaskList list) {
+        int index = Integer.parseInt(input.substring(7)) - 1;
+        return list.getTask(index);
+    }
+    
+    /**
+     * Parses the input command to identify the task to be deleted.
+     *
+     * @param input The user input string.
+     * @param list The TaskList containing all tasks.
+     * @return The Task to be deleted.
+     */
+    public Task getTaskToDelete(String input, TaskList list) {
+        int indexOfTaskToDelete = Integer.parseInt(input.substring(7)) - 1;
```

Comment:
I think here could write `int i` instead of that since its used in a small scope?